hack/make-rules/test-integration.sh

```diff
@@ -31,7 +31,7 @@ ALL_VERSIONS_CSV=$(IFS=',';echo "${KUBE_AVAILABLE_GROUP_VERSIONS[*]// /,}";IFS=$
 KUBE_TEST_API_VERSIONS="${KUBE_TEST_API_VERSIONS:-${ALL_VERSIONS_CSV}}"
 
 # Give integration tests longer to run by default.
-KUBE_TIMEOUT=${KUBE_TIMEOUT:--timeout 600s}
```

Comment:
@liggitt Feel a little uncomfortable about modifying this. But I have no better idea how to expand the integration-test timeout.

Reply:
agree. we don't want an e2e *or* an integration test that sleeps for 10 minutes.

it looks like you need two levers in NewTestDynamicClientBuilder:
* requestExpirationSeconds (set no lower than 600)
* maxExpiry (can set as low as you want)

When setting `Expiry` for a given token, set it to the min of 80% of the way to retTokenRequest.Status.ExpirationTimestamp.Time or time.Now().Add(maxExpiry)